Optik/Classes/Animation/AnimatableProperty.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  AnimatableProperty.swift
+//  Optik
+//
+//  Created by Htin Linn on 7/24/16.
+//  Copyright © 2016 Prolific Interactive. All rights reserved.
+//
+
+import UIKit
+
+/**
+ *  Defines an animatable property for the animation system.
+ */
+internal protocol AnimatableProperty {
+    
+    /// Underlying property type.
+    associatedtype PropertyType: VectorRepresentable
+    
+    /// Threshold value to use in determining if an animation is considered complete.
+    var threshold: CGFloat { get }
+    
+    /// Lens for reading and writing values.
+    var lens: Lens<UIView, PropertyType.InterpolatableType> { get }
+    
+}
+
+/**
+ *  View frame animatable property.
+ */
+internal struct ViewFrame: AnimatableProperty {
+    
+    typealias PropertyType = CGRect
+    
+    let threshold: CGFloat = 0.1
```

Comment:
I'm a little confused with this threshold, 0.1 here, and 0.01 in the implementation, what is the purpose of this property exaclty?